test/test_server.py

```diff
@@ -316,6 +317,14 @@ class TestHTTPS(tservers.HTTPProxTest, CommonMixin, TcpMixin):
     clientcerts = True
 
     def test_clientcert(self):
+        self.config.clientcerts = os.path.join(
+            tutils.test_data.path("data/clientcert"), "client.pem")
+        f = self.pathod("304")
+        assert f.status_code == 304
+        assert self.server.last_log()["request"]["clientcert"]["keyinfo"]
+
+    def test_clientcerts(self):
+        self.config.clientcerts = tutils.test_data.path("data/clientcert")
```

Comment:
This is the right way to implement it I think, but we should remove the `clientcerts = True` leftover in line 317 here and at https://github.com/mitmproxy/mitmproxy/blob/a978c6b9cee134d01f84216f3fbcf1ae9303d965/test/tservers.py#L133.
As a minor nitpick, please set `self.config.clientcerts` to `None` once the test is completed (you can use a `try: ... finally:` block).
Another nitpick: More expressive test names would be nice, e.g. `test_clientcert_file`/`test_clientcert_dir`  :-)